services/generic/alarm.md

```diff
@@ -1,41 +1,80 @@
-`alarm_appliance`   | program_started, program_inprogress, program_completed, replace_filter, set_temp_error, supplying_water, water_supply_err, boiling, boiling_err, washing, washing_err, rinsing, rinsing_err, draining, draining_err, spinning, spinning_err, drying, drying_err, fan_err, compressor_err |
-`alarm_burglar`     | intrusion, tamper_removed_cover, tamper_invalid_code, tamper_force_open, alarm_burglar, glass_breakage |
-`alarm_emergency`   | police, fire, medical                   |
-`alarm_fire`        | smoke, smoke_test                       |
-`alarm_gas`         | CO, CO2, combust_gas_detected, toxic_gas_detected, test, replace |
-`alarm_health`      | leaving_bed, sitting_on_bed, lying_on_bed, posture_change, sitting_on_bed_edge, alarm_health, volatile_organic_compound |
-`alarm_heat`        | overheat, temp_rise, underheat          |
-`alarm_lock`        | manual_lock, manual_unlock, rf_lock, rf_unlock, keypad_lock, keypad_unlock, tag_lock, tag_unlock, manual_not_locked, rf_not_locked, auto_locked, jammed, door_opened, door_closed, lock_failed | TODO: move to doorlock service
-`alarm_power`       | on, ac_on, ac_off, surge, voltage_drop, over_current, over_voltage, replace_soon, replace_now, charging, charged, charge_soon, charge_now | TODO: move to power_supply service
-`alarm_siren`       | inactive, siren_active                  |
-`alarm_system`      | hw_failure, sw_failure, hw_failure_with_code, sw_failure_with_code |
-`alarm_time`        | wakeup, timer_ended, time_remaining     |
-`alarm_water_valve` | valve_op, master_valve_op, valve_short_circuit, current_alarm, alarm_water_valve, master_valve_current_alarm |
-`alarm_water`       | leak, level_drop, replace_filter        |
-`alarm_weather`     | inactive, moisture                      |
-
-#### Interfaces
-
-Type | Interface            | Value type | Description
------|----------------------|------------|--------------------
-in   | cmd.alarm.get_report | str_map    | val = {"event": "tamper_removed_cover"}
-out  | evt.alarm.report     | str_map    | val = {"event": "tamper_removed_cover", "status": "activ"}
-in   | cmd.alarm.clear      | string     | val = event that should be cleared – device should send a new report after.
-
-Supported statuses: activ, deactiv. IMPORTANT: These are shorthands for "activated" and "deactivated", not typos.
-
-Interface cmd.alarm.get_report is used in Z-wave to clear last event for given event.
-
-Example message: [evt.sensor.report](json-v1/messages/examples/evt.alarm.report.json)
-
-#### Service props
-
-Name         | Value example           | Description
--------------|-------------------------|-------------
-`sup_events` | ["smoke", "smoke_test"] | supported events.
+# Alarm Services
+
+Alarm services are used to represent devices or functionalities responsible for detecting an undesired state and raising an alarm.
+Typical examples are SDCO (Smoke and Carbon Monoxide Detector), leak detector or burglar alarm.
```

Comment:
Alarm service creation was based on Command Class Alarm (deprecated name in z-wave), which is now called Command Class Notification. The "renaming" was done more than 5 years ago (circa).

With that being said, I want to point out that some alarms/notifications are not actually an "undesired state". Some are just notifications as a new name says.

Examples: 
1. Notification Type: Appliance, Notification Event: Boiling.
2. Notification Type: Power Managment, Notification Event: Battery is charging.
3. Notification Type: Access Control, Notification Event: Window/door is closed.

In our implementation, which includes just a part of Z-wave specification implementation, we actually have implemented mostly alarms. The question is, do we want to treat those Alarm Services as alarms, because most of it actually says about undesired state... Or we want to treat it as notifications.

I'm flexible, in this topic. Just wanted to point out the "Z-Wave facts" :)

Reply:
This is a very good commentary. Putting it in the post-morted doc to work on it further.